embulk-core/src/main/java/org/embulk/plugin/PluginClassLoader.java

```diff
@@ -254,6 +269,124 @@ private boolean isParentFirstPath(String name) {
+        // https://docs.oracle.com/javase/9/docs/api/java.xml.ws-summary.html
+        "javax.jws",
+        "javax.jws.soap",
+        "javax.xml.soap",
+        "javax.xml.ws",
+        "javax.xml.ws.handler",
+        "javax.xml.ws.handler.soap",
+        "javax.xml.ws.http",
+        "javax.xml.ws.soap",
+        "javax.xml.ws.spi",
+        "javax.xml.ws.spi.http",
+        "javax.xml.ws.wsaddressing",
+
+        // Module java.xml.bind : JAXB
+        // https://docs.oracle.com/javase/9/docs/api/java.xml.bind-summary.html
+        "javax.xml.bind",
+        "javax.xml.bind.annotation",
+        "javax.xml.bind.annotation.adapters",
+        "javax.xml.bind.attachment",
+        "javax.xml.bind.helpers",
+        "javax.xml.bind.util",
+
+        // Module java.activation : JAF
+        // https://docs.oracle.com/javase/9/docs/api/java.activation-summary.html
+        "javax.activation",
+
+        // Module java.xml.ws.annotation : Common Annotations
+        // https://docs.oracle.com/javase/9/docs/api/java.xml.ws.annotation-summary.html
+        "javax.annotation",
+
+        // Module java.corba : CORBA
+        // https://docs.oracle.com/javase/9/docs/api/java.corba-summary.html
+        "javax.activity",
+        "javax.rmi",
+        "javax.rmi.CORBA",
+        "org.omg.CORBA",
+        "org.omg.CORBA_2_3",
+        "org.omg.CORBA_2_3.portable",
+        "org.omg.CORBA.DynAnyPackage",
+        "org.omg.CORBA.ORBPackage",
+        "org.omg.CORBA.portable",
+        "org.omg.CORBA.TypeCodePackage",
+        "org.omg.CosNaming",
+        "org.omg.CosNaming.NamingContextExtPackage",
+        "org.omg.CosNaming.NamingContextPackage",
+        "org.omg.Dynamic",
+        "org.omg.DynamicAny",
+        "org.omg.DynamicAny.DynAnyFactoryPackage",
+        "org.omg.DynamicAny.DynAnyPackage",
+        "org.omg.IOP",
+        "org.omg.IOP.CodecFactoryPackage",
+        "org.omg.IOP.CodecPackage",
+        "org.omg.Messaging",
+        "org.omg.PortableInterceptor",
+        "org.omg.PortableInterceptor.ORBInitInfoPackage",
+        "org.omg.PortableServer",
+        "org.omg.PortableServer.CurrentPackage",
+        "org.omg.PortableServer.POAManagerPackage",
+        "org.omg.PortableServer.POAPackage",
+        "org.omg.PortableServer.portable",
+        "org.omg.PortableServer.ServantLocatorPackage",
+        "org.omg.SendingContext",
+        "org.omg.stub.java.rmi",
+
+        // Module java.transaction : JTA
+        // https://docs.oracle.com/javase/9/docs/api/java.transaction-summary.html
+        "javax.transaction",
+    };
+
+    private static Set<String> JEP_320_PACKAGES =
+            Collections.unmodifiableSet(new HashSet<String>(Arrays.asList(JEP_320_PACKAGES_ARRAY)));
+
     private final List<String> parentFirstPackagePrefixes;
     private final List<String> parentFirstResourcePrefixes;
+
+    private boolean hasJep320LoggedWithStackTrace;
```

Comment:
A matter of preference, but it seems to me that having this flag is not very advantageous -- such logs are not likely to be observed frequently and it's fine to have them verbose. Also it may help to find transitive dependency issues faster.

Reply:
When one class is loaded from one set of feature (e.g. JAXB), that is typically not the only one class. Related classes are loaded one after another in typical use-cases. So, when one class is logged, a lot more are logged. Most of them are not very informative, though.
https://github.com/embulk/embulk/pull/1271/files#diff-b05395f45e2d8aad0e598eaf4185c738R278

Also, the log is dumped almost always when the affected plugin is used, even though the user (including ourselves) is aware of that.